cluster-autoscaler/core/scaledown/unneeded/nodes.go

```diff
@@ -117,21 +117,20 @@ func (n *Nodes) Drop(node string) {
 // RemovableAt returns all nodes that can be removed at a given time, divided
 // into empty and non-empty node lists, as well as a list of nodes that were
 // unneeded, but are not removable, annotated by reason.
-func (n *Nodes) RemovableAt(context *context.AutoscalingContext, ts time.Time, resourcesLeft resource.Limits, resourcesWithLimits []string, as scaledown.ActuationStatus) (empty, needDrain []*apiv1.Node, unremovable []*simulator.UnremovableNode) {
+func (n *Nodes) RemovableAt(context *context.AutoscalingContext, ts time.Time, resourcesLeft resource.Limits, resourcesWithLimits []string, as scaledown.ActuationStatus) (empty, needDrain []simulator.NodeToBeRemoved, unremovable []*simulator.UnremovableNode) {
```

Comment:
Since you're already changing the function signature to return `simulator.NodeToBeRemoved`, I think it makes sense to just put empty & non-empty on a single list and let the caller worry about splitting them.

Reply:
I don't have strong opinion on this, but both callers of this method require distinction between empty and non-empty nodes to remove. Does it make sense then to merge the lists into one?

Reply:
I think this results in a slightly cleaner interface. Don't have a too strong opinion on this either though, so up to you.

Reply:
If there is no strong opinion here, I would leave it as is, given we need the distinction.